src/main/scala/Shingler.scala

```diff
@@ -0,0 +1,50 @@
+package shingler
+
+/** Bundles Shingler finding and hashing functionalities
+  *
+  * @param shingle_len the length (e.g. in characters) in which to split the document
+  * @param hash_bins the number of bins into which to map the shingle hashes
+  */
+class Shingler(shingle_len: Int, hash_bins : Int) {
+
+  /** Split document into shingles
+    * @param doc Document which needs to be split
+    * @return Set of shingles found in document
+    */
+  def getShingles(doc : String) : Set[String] = {
```

Comment:
In general, functions could be rewritten as "def getShingles(doc: String): Set[String] = blabla. One-liners do not need braces, and ":" should stick to the left.

Reply:
Oh nice!